src/StructuredLogViewer.Core/NodeQueryMatcher.cs

```diff
@@ -69,9 +56,23 @@ public NodeQueryMatcher(string query, IEnumerable<string> stringTable)
 
                 if (word.StartsWith("under(", StringComparison.OrdinalIgnoreCase) && word.EndsWith(")"))
                 {
-                    word = word.Substring(6, word.Length - 7);
+                    var nodeIndexText = word.Substring(7, word.Length - 8);
                     Words.RemoveAt(i);
-                    UnderMatcher = new NodeQueryMatcher(word, stringTable);
+                    if (int.TryParse(nodeIndexText, out var nodeIndex))
+                    {
+                        UnderNodes.Add(nodeIndex);
+                    }
+                    continue;

```

Comment:
we still need to have an UnderMatcher if the query is not a node index.

You can have queries like `$task Csc under($project Foo)`, so UnderMatcher recursively evaluates the expression in parentheses `$project Foo`